lighthouse-core/report/html/renderer/util.js

```diff
@@ -280,6 +280,79 @@ class Util {
     return parts.join(' ');
   }
 
+  /**
+   * Split a string by markdown code spans (enclosed in `backticks`), splitting
+   * into segments that were enclosed in backticks (marked as `isCode === true`)
+   * and those that outside the backticks (`isCode === false`).
+   * @param {string} text
+   * @return {Array<{isCode: true, codeText: string}|{isCode: false, plainText: string}>}
+   */
+  static splitMarkdownCodeSpans(text) {
+    /** @type {Array<{isCode: true, codeText: string}|{isCode: false, plainText: string}>} */
+    const segments = [];
+
+    // Split on backticked code spans.
+    const parts = text.split(/`(.*?)`/g);
+    for (let i = 0; i < parts.length; i ++) {
+      const text = parts[i];
+
+      // Empty strings are an artifact of splitting, not meaningful.
+      if (!text) continue;
+
+      // Alternates between plain text and code segments.
+      if (i % 2 === 0) {
+        segments.push({
+          isCode: false,
+          plainText: text,
+        });
+      } else {
+        segments.push({
+          isCode: true,
+          codeText: text,
+        });
+      }
+    }
+
+    return segments;
+  }
+
+  /**
+   * Split a string on markdown links (e.g. [some link](https://...)) into
+   * segments of plain text that weren't part of a link (marked as
+   * `isLink === false`), and segments with text content and a URL that did make
+   * up a link (marked as `isLink === true`).
+   * @param {string} text
+   * @return {Array<{isLink: true, linkText: string, linkHref: string}|{isLink: false, plainText: string}>}
+   */
+  static splitMarkdownLink(text) {
+    /** @type {Array<{isLink: true, linkText: string, linkHref: string}|{isLink: false, plainText: string}>} */
+    const segments = [];
+
+    const parts = text.split(/\[([^\]]+?)\]\((https?:\/\/.*?)\)/g);
+    while (parts.length) {
+      // Pop off the same number of elements as there are capture groups.
```

Comment:
```suggestion
      // Shift off the same number of elements as there are capture groups.
```

;)

largely a pedantic suggestion and pre-existing, so can reject if you'd like